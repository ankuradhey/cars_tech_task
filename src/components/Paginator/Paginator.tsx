import React, { FC } from "react";
import { Link } from "../Link";
import { Text } from "../Text";

export const Paginator: FC = ({ currentPage, totalPages, navigateToPage }) => {
    return (
        <div className="d-flex justify-content-center">
            <div className="p-2 bd-highlight">
                {currentPage > 1 ? (
                    <Link onClick={() => navigateToPage(1)} data-testid="first-page">
                        First
                    </Link>
                ) : (
                    <Text size="small" data-testid="first-page">
                        First
                    </Text>
                )}
            </div>
            <div className="p-2 bd-highlight">
                {currentPage > 1 ? (
                    <Link
                        onClick={() => navigateToPage(currentPage - 1)}
                        data-testid="previous-page"
                    >
                        Previous
                    </Link>
                ) : (
                    <Text size="small" data-testid="previous-page">
                        Previous
                    </Text>
                )}
            </div>
            <div className="p-2 bd-highlight">
                <Text size="small">
                    Page {currentPage} of {totalPages}
                </Text>
            </div>
            <div className="p-2 bd-highlight">
                {currentPage < totalPages ? (
                    <Link onClick={() => navigateToPage(currentPage + 1)} data-testid="next-page">
                        Next
                    </Link>
                ) : (
                    <Text size="small" data-testid="next-page">
                        Next
                    </Text>
                )}
            </div>
            <div className="p-2 bd-highlight">
                {currentPage < totalPages ? (
                    <Link onClick={() => navigateToPage(totalPages)} data-testid="last-page">
                        Last
                    </Link>
                ) : (
                    <Text size="small" data-testid="last-page">
                        Last
                    </Text>
                )}
            </div>
        </div>
    );
};
