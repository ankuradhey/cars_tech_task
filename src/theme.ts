import { DefaultTheme } from "styled-components";

export const defaultTheme: DefaultTheme = {
    borderRadius: "2px",
    palette: {
        common: {
            black: "#4A4A4A",
            white: "#EDEDED",
        },
        primary: {
            main: "#EA7F28",
            contrastText: "#ffffff",
        },
        secondary: {
            main: "#D37324",
            contrastText: "#ffffff",
        },
        tertiary: {
            main: "#ffffff",
            contrastText: "#EDEDED",
        },
    },
    gutterSize: {
        sm: "8px",
        md: "12px",
        lg: "24px",
    },
};
