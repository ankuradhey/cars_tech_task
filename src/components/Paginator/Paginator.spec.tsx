import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../../theme";
import { Paginator } from "./Paginator";

describe("Paginator Component", () => {
    let component, navigateToPage;
    beforeEach(() => {
        navigateToPage = jest.fn(() => {});
        component = render(
            <ThemeProvider theme={defaultTheme}>
                <Paginator currentPage={1} totalPages={10} navigateToPage={navigateToPage} />
            </ThemeProvider>
        );
    });

    it("Should render Paginator component", () => {
        let { container, getByText } = component;

        expect(getByText("First")).toBeInTheDocument();
        expect(getByText("Next")).toBeInTheDocument();
        expect(getByText("Previous")).toBeInTheDocument();
        expect(getByText("Last")).toBeInTheDocument();
        expect(getByText("Page 1 of 10")).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });

    it("Should navigate to other pages", () => {
        let { getByTestId } = component;

        fireEvent.click(
            getByTestId("next-page"),
            new MouseEvent("click", {
                bubbles: true,
                cancelable: true,
            })
        );

        expect(navigateToPage).toHaveBeenCalledWith(2);
    });
});
