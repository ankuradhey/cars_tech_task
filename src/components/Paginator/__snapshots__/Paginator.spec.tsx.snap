// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator Component Should render Paginator component 1`] = `
<div
  class="d-flex justify-content-center"
>
  <div
    class="p-2 bd-highlight"
  >
    <div
      class="sc-gsTCUz hhHEdt"
      data-testid="first-page"
    >
      First
    </div>
  </div>
  <div
    class="p-2 bd-highlight"
  >
    <div
      class="sc-gsTCUz hhHEdt"
      data-testid="previous-page"
    >
      Previous
    </div>
  </div>
  <div
    class="p-2 bd-highlight"
  >
    <div
      class="sc-gsTCUz hhHEdt"
    >
      Page 
      1
       of 
      10
    </div>
  </div>
  <div
    class="p-2 bd-highlight"
  >
    <button
      class="sc-bdfBwQ jSTqUA"
      data-testid="next-page"
    >
      Next
    </button>
  </div>
  <div
    class="p-2 bd-highlight"
  >
    <button
      class="sc-bdfBwQ jSTqUA"
      data-testid="last-page"
    >
      Last
    </button>
  </div>
</div>
`;
